hydromt/data_source/geodataframe.py

```diff
@@ -0,0 +1,198 @@
+"""Generic DataSource for GeoDataFrames."""
+
+from datetime import datetime
+from logging import Logger, getLogger
+from os.path import basename, splitext
+from typing import ClassVar, List, Literal, Optional
+
+import geopandas as gpd
+from pydantic import Field
+from pyproj import CRS
+from pyproj.exceptions import CRSError
+from pystac import Asset as StacAsset
+from pystac import Catalog as StacCatalog
+from pystac import Item as StacItem
+from pystac import MediaType
+
+from hydromt._typing import Bbox, ErrorHandleMethod, Geom, NoDataStrategy, TotalBounds
+from hydromt.data_adapter.geodataframe import GeoDataFrameAdapter
+from hydromt.driver.geodataframe_driver import GeoDataFrameDriver
+
+from .data_source import DataSource
+
+logger: Logger = getLogger(__name__)
+
+
+class GeoDataFrameSource(DataSource):
```

Comment:
Should this example not give errors for unknown arguments like `foo="bar"`

```py 
source = GeoDataFrameSource(
    name="test",
    uri="points.geojson",
    driver={"name": "pyogrio"},
    data_type="GeoDataFrame", # this is not used / needed
    bar="foo", # shouldn't this raise an error?
)
```

Reply:
You are correct, I have added model_config extra = strict: https://docs.pydantic.dev/latest/concepts/strict_mode/